packages/clay-modal/src/Provider.tsx

```diff
@@ -0,0 +1,91 @@
+/**
+ * © 2019 Liferay, Inc. <https://liferay.com>
+ *
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+import ClayModal from './Modal';
+import React, {createContext, useReducer} from 'react';
+import {Size, Status} from './types';
+
+enum Action {
+	Close = 0,
+	Open = 1,
+}
+
+interface IProps {
+	spritemap?: string;
+}
+
+type TState = {
+	body: React.ReactElement | React.ReactText;
+	footerFirst?: React.ReactElement;
+	footerLast?: React.ReactElement;
```

Comment:
Instead of first and last, would it work to add an array for footer? It'd give a bit more flexibility in the future.

I saw we have `first` `middle` and `last` in our Footer component. So maybe an array would make sense, primarily thinking if in the future we add more areas to footer, we then have `footerNextToLast`. This may be a pre-mature optimization, but think it'd make it at least a bit more flexible.

Reply:
Hmm, I think that might make sense. But you claim to be an Array on the Provider side and pass it on to `ClayModal.Footer`? Considering the first element of the array is `first`, second `middle` and third `last`.

Reply:
Yeah, I would think `footer?: Array<React.ReactElement>` where the first index is the left most and the end is the right most element. Not super important, was just a thought of mine when thinking of "how does this sort of API scale."

Reply:
Oh yeah, sounds good.